pkg/sdk/testint/security_integrations_gen_integration_test.go

```diff
@@ -24,63 +33,63 @@ func TestInt_SecurityIntegrations(t *testing.T) {
-	WYxLNx9pbaPIORAJ1KEC3hvE4strHJPlQddCYSsXDhIOUTUd71JvR26DHiYQ82TO
-	3wpXHhYdWYZbMjrDDAz0PwdTXyFBuTZxdlTFTxX2lXAE33OsdAFt+oi7JTQh248k
-	0+lmQdhQrSrzhM3WwwuYTEKQVoa2xvWajgqbo7iu2iadWkrxUx/5bjFc5kXej6j7
-	PhfG6C4ddUpAISJhmEViuXq4nVxe0Vk3Efo2
-	`
-
-	createSAML2Integration := func(t *testing.T, siID sdk.AccountObjectIdentifier, with func(*sdk.CreateSAML2SecurityIntegrationRequest)) {
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(1658),
+		Subject: pkix.Name{
+			Organization: []string{"Company, INC."},
+		},
+		NotAfter:    time.Now().AddDate(10, 0, 0),
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+		KeyUsage:    x509.KeyUsageDigitalSignature,
+	}
+
+	caPrivKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err)
+
+	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPrivKey.PublicKey, caPrivKey)
+	require.NoError(t, err)
+
+	certPEM := new(bytes.Buffer)
+	err = pem.Encode(certPEM, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: caBytes,
+	})
+	require.NoError(t, err)
+
+	cert := strings.TrimPrefix(certPEM.String(), "-----BEGIN CERTIFICATE-----\n")
+	cert = strings.TrimSuffix(cert, "-----END CERTIFICATE-----\n")
+
+	createSAML2Integration := func(t *testing.T, siID sdk.AccountObjectIdentifier, issuer string, with func(*sdk.CreateSaml2SecurityIntegrationRequest)) {
 		t.Helper()
 		_, err := client.ExecForTests(ctx, "ALTER ACCOUNT SET ENABLE_IDENTIFIER_FIRST_LOGIN = true")
 		require.NoError(t, err)
 
-		saml2Req := sdk.NewCreateSAML2SecurityIntegrationRequest(siID, false, "test", "https://example.com", "Custom", x509)
+		saml2Req := sdk.NewCreateSaml2SecurityIntegrationRequest(siID, false, issuer, "https://example.com", "Custom", cert)
 		if with != nil {
 			with(saml2Req)
 		}
-		err = client.SecurityIntegrations.CreateSAML2(ctx, saml2Req)
+		err = client.SecurityIntegrations.CreateSaml2(ctx, saml2Req)
 		require.NoError(t, err)
 		cleanupSecurityIntegration(t, siID)
 	}
 
-	createSCIMIntegration := func(t *testing.T, siID sdk.AccountObjectIdentifier, with func(*sdk.CreateSCIMSecurityIntegrationRequest)) {
+	createSCIMIntegration := func(t *testing.T, siID sdk.AccountObjectIdentifier, with func(*sdk.CreateScimSecurityIntegrationRequest)) *sdk.SecurityIntegration {
 		t.Helper()
-		roleID := sdk.NewAccountObjectIdentifier("GENERIC_SCIM_PROVISIONER")
-		err := client.Roles.Create(ctx, sdk.NewCreateRoleRequest(roleID).WithIfNotExists(true))
-		require.NoError(t, err)
-		t.Cleanup(func() {
-			err = client.Roles.Drop(ctx, sdk.NewDropRoleRequest(roleID))
-			assert.NoError(t, err)
-		})
-		currentRole := testClientHelper().Context.CurrentRole(t)
-		err = client.Roles.Grant(ctx, sdk.NewGrantRoleRequest(roleID, sdk.GrantRole{Role: sdk.Pointer(sdk.NewAccountObjectIdentifier(currentRole))}))
-		require.NoError(t, err)
+		role, roleCleanup := testClientHelper().Role.CreateRoleWithRequest(t, sdk.NewCreateRoleRequest(sdk.NewAccountObjectIdentifier("GENERIC_SCIM_PROVISIONER")).WithOrReplace(true))
```

Comment:
I didn't point this out earlier, but it's better to use RandomAccountObjectIdentifier or other new random id variations for test ids

Reply:
These are the predefined ones as already answered in comment https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2799#discussion_r1603020981

Reply:
Fixed.

Reply:
Fixed.